### PR TITLE
Add skip_on_ci() function

### PR DESCRIPTION
**Pull request by mbjoseph**
This PR adds a `skip_on_ci()` function that uses the `CI` environment variable to skip tests running on continuous integration systems, and would solve https://github.com/r-lib/testthat/issues/825. 

* Edit: Generalized from `skip_on_circleci()` to `skip_on_ci()`

### CODE REVIEW

File: R/skip.R [hunk 1]
```diff
@@ -163,6 +166,17 @@ skip_on_appveyor <- function() {
   skip("On Appveyor")
 }
 
+#' @export
+#' @rdname skip
+skip_on_ci <- function() {
+  ci <- Sys.getenv("CI")
+  if (!any(identical(ci, "true"), identical(ci, "True"))) {
```
Comment (jimhester) [inline R/skip.R]:
```suggestion
  if (!isTRUE(as.logical(Sys.getenv("CI")))) {
```

Comment (jimhester) [reply R/skip.R]: This is a potential shorter / more robust solution.

``` r
isTRUE(as.logical("true"))
#> [1] TRUE
isTRUE(as.logical("True"))
#> [1] TRUE
isTRUE(as.logical("False"))
#> [1] FALSE
isTRUE(as.logical(""))
#> [1] FALSE
isTRUE(as.logical(NULL))
#> [1] FALSE
isTRUE(as.logical(NA))
#> [1] FALSE
```

<sup>Created on 2018-12-04 by the [reprex package](https://reprex.tidyverse.org) (v0.2.1)</sup>

**Comment by mbjoseph** [comment]
Thank you @jimhester! I swapped in `isTRUE` as suggested, and added a NEWS bullet in 7ff05d4

**Comment by hadley** [comment]
Thanks!

**Pull request merged**